l10n_ch_fds_upload_sepa/wizards/fds_postfinance_file_pain000.py

```diff
@@ -0,0 +1,43 @@
+from odoo import models, api
+
+import base64
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+class FdsPostfinanceFilePain000(models.Model):
+    _inherit = 'fds.postfinance.file'
+
+    @api.multi
+    def import2bankStatements(self):
+        account_pain000 = self.env['account.pain000.parser']
+        for pf_file in self:
+            values = {'data_file': pf_file.data}
+            bs_import_obj = self.env['account.bank.statement.import']
+            bank_wiz_imp = bs_import_obj.create(values)
+
+            decoded_file = base64.b64decode(bank_wiz_imp.data_file)
+            result = account_pain000.parse(decoded_file)
+            if result[0]:
+                if result[1] is not None and result[1].id:
+                    # Link the payment order to the file import.
+                    pf_file.payment_order = result[1].id
+                    # Attach the file to the payment order.
+                    self.env['ir.attachment'].create({
+                            'datas_fname': pf_file.filename,
+                            'res_model': 'account.payment.order',
+                            'datas': bank_wiz_imp.data_file,
+                            'name': pf_file.filename,
+                            'res_id': result[1].id})
+
+                pf_file.write({
+                    'state': 'done',
+                    'data': bank_wiz_imp.data_file
+                })
+
+                _logger.info("[OK] import file '%s' as pain.000",
+                             pf_file.filename)
+                return True
```

Comment:
You should not return in the loop as this will break the function and following records in self will never be processed. Instead of this I encourage you to use the recordsets method to filter those you processed and at the end only call super on the records that you didn't process. Here is pseudo code to help you :
```
pain_files = self.env[self._name]  # Empty recordset of postfinance.file
for pf_file in self:
    ...
    # if pf_file is treated as pain file (instead of return True line of code)
    pain_files += pf_file
return super(FdsPostfinanceFilePain000, self - pain_files).import2bankStatements()
```